query-parser/src/parse.rs

```diff
@@ -303,11 +305,28 @@ def_parser!(Query, aggregate, Aggregate, {
          })
 });
 
+def_parser!(Query, pull_concrete_attribute_ident, PullConcreteAttribute, {
```

Comment:
I would have expected to be expressed as an optional `:as foo`, but I guess this works as well.  The multiple try blocks should be a little less efficient than the optional block, I think.

Reply:
Yep — Datomic's grammar is wrong. I will fix.